src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -885,11 +885,12 @@ export class SavedObjectsRepository {
       );
     }
 
-    const types = type
-      ? Array.isArray(type)
-        ? type
-        : [type]
-      : Array.from(typeToNamespacesMap!.keys());
+    const types =
+      type && type.length > 0
+        ? Array.isArray(type)
+          ? type
+          : [type]
+        : Array.from(typeToNamespacesMap!.keys());
```

Comment:
This change is causing the CI error, because the Spaces SOC wrapper filters out the "space" type from the `type` string array: 

https://github.com/elastic/kibana/blob/4681a80317176543535bfdb2188d94aa40d741fc/x-pack/plugins/spaces/server/saved_objects/spaces_saved_objects_client.ts#L160-L162

The Spaces SOC wrapper executes _before_ the Security SOC wrapper. So in the CI test, the repository has `type: [], typeToNamespacesMap: undefined`, and the change here tries to get the keys from an undefined map.

I tried removing this change and the test still passes, so I don't think we need this change, only the other one. WDYT?

Reply:
I am all for it ;)